spec/yaml/swiftnav/sbp/system.yaml

```diff
@@ -134,6 +134,83 @@ definitions:
                       - 0: System Healthy
                       - 1: An error has occurred
 
+ - SubSystemReport:
+    short_desc: Sub-system Status report
+    embedded_type: true
+    desc: |
+      Report the general and specific state of a sub-system
+    fields:
+        - component:
+            type: u16
+            desc: Identity of reporting subsystem
+            fields:
+              - 0-15:
+                  desc: Subsystem
+                  values:
+                      - 0: Primary GNSS Antenna
+                      - 1: Measurement Engine
+                      - 2: Corrections Client
+                      - 3: Differential GNSS Engine
+                      - 4: CAN
+                      - 5: Wheel Odometry
+                      - 6: Sensor Fusion Engine
+        - generic:
+            type: u8
+            desc: Generic form status report
+            fields:
+              - 0-7:
+                  desc: Generic
+                  values:
+                      - 0: OK/Nominal
+                      - 1: Initialising
+                      - 2: Unknown
+                      - 3: Degraded
+                      - 4: Unusable
+        - specific:
+            type: u8
+            desc: Subsystem specific status code
+
+ - MSG_STATUS_REPORT:
+    id: 0xFFFE
+    short_desc: Status report message
+    desc: |
+        The status report is sent periodically to inform the host
+        or other attached devices that the system is running. It is
+        used to monitor system malfunctions. It contains status
+        reports that indicate to the host the status of each sub-system and
+        whether it is operating correctly.
+
+        Interpretation of the subsystem specific status code is product dependent.
+        Refer to product documentation for details.
+    fields:
+        - reporting_system:
+            type: u16
+            desc: Identity of reporting system
+            fields:
+              - 0-15:
+                 desc: System
+                 values:
+                     - 0: Starling
+                     - 1: Precision GNSS Module (PGM)
+        - sbp_version:
+            type: u16
+            desc: SBP protocol version
+            fields:
+              - 8-16:
+                  desc: SBP major protocol version number
+              - 0-7:
+                  desc: SBP minor protocol version number
+        - sequence:
+            type: u32
+            desc: Increments on each status report sent
+        - uptime:
+            type: u32
+            desc: Number of milliseconds since system start-up
```

Comment:
Is this mean to be time since starling startup or system? 2^32ms is only ~7weeks so wrapping isn't out of the questions particularly if the value comes from system time and starling was started some time after system start. Is it possible to use u64 here?

Reply:
@dgburr What is your use case for wanting the system uptime in milliseconds?  This was intended to cover the requirement brought up by @switanis to report system uptime, currently reported in seconds in a MSG_LOG output.

To me, it seems interesting to know how many minutes, hours or days the system has been running, I can't imagine caring about sub-second resolution.

Reply:
@gsmcmullin: I was not involved in the definition of the uptime message (it seems to originate from the Amazon project)